airbyte-cdk/python/airbyte_cdk/sources/utils/catalog_helpers.py

```diff
@@ -12,11 +14,11 @@ def coerce_catalog_as_full_refresh(catalog: AirbyteCatalog) -> AirbyteCatalog:
         """
         Updates the sync mode on all streams in this catalog to be full refresh
         """
-        coerced_catalog = catalog.copy()
+        coerced_catalog = copy(catalog)
         for stream in catalog.streams:
             stream.source_defined_cursor = False
             stream.supported_sync_modes = [SyncMode.full_refresh]
             stream.default_cursor_field = None
 
         # remove nulls
-        return AirbyteCatalog.parse_raw(coerced_catalog.model_dump_json(exclude_unset=True, exclude_none=True))
+        return AirbyteCatalogSerializer.load(AirbyteCatalogSerializer.dump(coerced_catalog))
```

Comment:
@artem1205 can you add a testcase showing this is removing the null values?

Reply:
this _helper_ was used only in `singer` and not relevant anymore.
Removed in https://github.com/airbytehq/airbyte/pull/44444/commits/d574c8461bb6d969d1f1cbe26a5e3f9408ccefec